modules/standard/SysError.chpl

```diff
@@ -52,6 +52,164 @@ class SystemError : Error {
   proc writeThis(f) {
     try! f.write(msg);
   }
+
+  proc type fromSyserr(err: syserr, msg: string = "") {
+    var strerror_err: err_t = ENOERR;
+    var errstr : c_string = sys_strerror_syserr_str(err, strerror_err);
+    var err_msg: string   = errstr:string;
+    if !msg.isEmptyString() then
+      err_msg = errstr:string + " (" + msg + ")";
+
+    if err == EAGAIN || err == EALREADY || err == EWOULDBLOCK || err == EINPROGRESS {
+      return new BlockingIOError(err, err_msg);
+    } else if err == ECHILD {
+      return new ChildProcessError(err_msg);
+    } else if err == ENOENT {
+      return new FileNotFoundError(err_msg);
+    } else if err == EPIPE || err == ESHUTDOWN {
+      return new BrokenPipeError(err, err_msg);
+    } else if err == ECONNABORTED {
+      return new ConnectionAbortedError(err_msg);
+    } else if err == ECONNREFUSED {
+      return new ConnectionRefusedError(err_msg);
+    } else if err == ECONNRESET {
+      return new ConnectionResetError(err_msg);
+    } else if err == EEXIST {
+      return new FileExistsError(err_msg);
+    } else if err == ENOENT {
+      return new FileNotFoundError(err_msg);
+    } else if err == EINTR {
+      return new InterruptedError(err_msg);
+    } else if err == EISDIR {
+      return new IsADirectoryError(err_msg);
+    } else if err == ENOTDIR {
+      return new NotADirectoryError(err_msg);
+    } else if err == EACCES || err == EPERM {
+      return new PermissionError(err, err_msg);
+    } else if err == ESRCH {
+      return new ProcessLookupError(err_msg);
+    } else if err == ETIMEDOUT {
+      return new TimeoutError(err_msg);
+    }
```

Comment:
Chapel's I/O system adds 3 "system" error codes. Of course these aren't thrown by system calls, but they can be raised doing I/O. They are:
 * EEOF - An error code indicating the end of file has been reached
 * ESHORT - An error code indicating that the end of file or the end of the input was reached before the requested amount of data could be read.
 * EFORMAT - An error code indicating a format error; for example when reading a quoted string literal, this would be returned if we never encountered the opening quote.

It seems to me that each of these should get its own subclass of SystemError.
Of course we will have to make up names for them, but these will almost certainly each be multi-word names to fit in with the others. Maybe
 * EEOF -> EndOfFileError
 * ESHORT -> UnexpectedEndOfFileError
 * EFORMAT -> BadFormatError